pkg/adapter/v2/cloudevents.go

```diff
@@ -105,6 +107,7 @@ func NewCloudEventsClientCRStatus(env EnvConfigAccessor, reporter source.StatsRe
 }
 
 type ClientConfig struct {
+	Context             context.Context
```

Comment:
We shouldn't add a field for the context, because a function/method which depends on a context, should be get it directly via a ctx argument. 
But as you need it to setup the tokenProvider, you can let the user pass the TokenProvider here instead (and initialize it in adapter/v2/main.go)

Reply:
opps, forgot to remove that. I pass that for debugging purpose. In order to use the logger, I have to pass in the context. I have updated it.